qiskit_ibm_runtime/sampler.py

```diff
@@ -117,8 +118,12 @@ def run(self, pubs: Iterable[SamplerPubLike], *, shots: int | None = None) -> Ru
                 "will be empty. Did you mean to add measurement instructions?",
                 UserWarning,
             )
-
-        return self._run(coerced_pubs)  # type: ignore[arg-type]
+        # Unpack pubs back to Pub-like tuples for serialization
+        unpacked_pubs = [
+            (pub.circuit, pub.parameter_values.as_array(pub.circuit.parameters), pub.shots)
```

Comment:
Does the simple `as_array` work? Since parameter values can be a dictionary, the values are not necessarily in the same order as the circuit parameters.

Reply:
Yes, this is the intended API for `BindingsArray` to convert to numpy array on correctly ordered parameters